test_tag_generator.rb

```diff
@@ -0,0 +1,13 @@
+class TestTagGenerator
+
+  def check_for_tag?(check_tag_type, tags_for_this_commit)
+    tags_for_this_commit.any? {|tag| /^#{check_tag_type}-v\d+.\d+.\d*$/ =~ tag}
+  end
+
+  def next_tag(previous_tag_type, next_tag_type, tags_for_this_commit)
+    tags_for_this_commit.select {|tag| /^#{previous_tag_type}-v\d+.\d+.\d*$/ =~ tag}
```

Comment:
what would happen if there are different tags on the same commit. so let's say that there is dev-v0.0.1 and dev-v0.0.2 and you call 
```
next_tag(dev, test, [dev-v0.0.1, dev-v0.0.2])
```